layout21converters/src/gds2ser.rs

```diff
@@ -0,0 +1,194 @@
+//!
+//! # GDSII to Alternate Serialization Conversion CLI
+//! 
+//! Converts a [`gds21::GdsLibrary`] to any of the supported alternate serialization formats.
+//!
+
+use clap::Parser;
+use layout21utils::SerializationFormat;
+use std::error::Error;
+
+// => The doc-comment on `ProgramOptions` here is displayed by the `clap`-generated help docs => 
+
+/// GDSII to Alternate Serializer Converter
+#[derive(Parser)]
+pub struct ProgramOptions {
+    /// GDS Input File
+    #[clap(short = 'i', long, default_value = "")]
+    gds: String,
+    /// Output Format. One of ("json", "yaml", "toml")
+    #[clap(short = 'f', long, default_value = "")]
+    fmt: String,
+    /// Output File
+    #[clap(short = 'o', long, default_value = "")]
+    out: String,
+    /// Verbose Output Mode
+    #[clap(short, long)]
+    verbose: bool,
+}
+
+pub fn main() -> Result<(), Box<dyn Error>> {
+    let options = ProgramOptions::parse();
+    _main(&options)
+}
+
+/// Parse the `fmt` string into a [`SerializationFormat`].
+/// FIXME: make this a [`FromStr`] impl for [`SerializationFormat`] instead.
+fn parse_format(format: &str) -> Result<SerializationFormat, Box<dyn Error>> {
+    match format {
+        "json" => Ok(SerializationFormat::Json),
+        "yaml" => Ok(SerializationFormat::Yaml),
+        "toml" => Err(format!("TOML is not yet supported, see https://github.com/dan-fritchman/Layout21/issues/33").into()),
```

Comment:
seems more straightforward to just make this do what it will have to eventually do when TOML is fixed, and then fix #33

Reply:
Definitely agree it's more straightforward. 

Problem is the error you get when just allowing TOML to fail natively is less than ideal, as described in #33 and in  https://github.com/toml-rs/toml/issues/396

Reply:
Probably worth adding -  
it's not entirely clear that #33 is actually fixable.

Reply:
I think `gds2markup` (not "markdown") seems probably the best name the multi-format program currently called `gds2ser`? 

Prefer anything else?